### PR TITLE
Deprecate `describeEnum`.

### PR DESCRIPTION
**Pull request by bernaferrari**
Tied to https://github.com/flutter/flutter/pull/125016

### CODE REVIEW

File: src/release/breaking-changes/describeEnum-migration.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+<!-- skip -->
+```dart
+enum MyEnum { paper, rock }
+
+print(MyEnum.paper.name); // output: paper
+```
+
+## Timeline
+
+Landed in version: TBD<br>
```
Comment (sfshaza2) [inline src/release/breaking-changes/describeEnum-migration.md:52]: Please fill out this TBD with the info on when this landed.

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
Hi, @bernaferrari! Thanks for this PR! Can you also add this breaking change to the index file?

**Review by sfshaza2 — APPROVED** [review]
lgtm

**Comment by bernaferrari** [comment]
(should this be merged? The original code/official change hasn't landed yet, might take a few days/weeks still)

**Pull request merged**